src/kbmod/reprojection_utils.py

```diff
@@ -55,3 +56,54 @@ def correct_parallax(coord, obstime, point_on_earth, guess_distance):
     ).transform_to(ICRS())
 
     return answer
+
+
+def fit_barycentric_wcs(original_wcs, width, height, distance, obstime, point_on_earth, npoints=10):
+    """Given a ICRS WCS and an object's distance from the Sun,
+        return a new WCS that has been corrected for parallax motion.
+
+    Attributes
+    ----------
+    original_wcs : `astropy.wcs.WCS`
+        The image's WCS.
+    width : `int`
+        The image's width (typically NAXIS1).
+    height : `int`
+        The image's height (typically NAXIS2).
+    distance : `float`
+        The distance of the object from the sun, in AU.
+    obstime : `astropy.time.Time` or `string`
+        The observation time.
+    point_on_earth : `astropy.coordinate.EarthLocation`
+        The location on Earth of the observation.
+    npoints : `int`
+        The number of randomly sampled points to use during the WCS fitting.
+            Typically, the more points the higher the accuracy. The four corners
+            of the image will always be included, so setting npoints = 0 will mean
+            just using the corners.
+
+    Returns
+    ----------
+    An `astropy.wcs.WCS` representing the original image in "Explicity Barycentric Distance" (EBD)
+        space, i.e. where the points have been corrected for parallax.
+    """
+    sampled_x_points = np.array([0, 0, width, width])
+    sampled_y_points = np.array([0, height, height, 0])
+    if npoints > 0:
+        sampled_x_points = np.append(sampled_x_points, np.random.rand(npoints) * width)
+        sampled_y_points = np.append(sampled_y_points, np.random.rand(npoints) * height)
```

Comment:
I think it would be worth adding an optional parameter to seed the random number generator so we can get consistent random points for testing or debugging. You'd need to change this to use a numpy random number generator.

This can be a followup PR.